### PR TITLE
Integrating Ardin MB example with BoB robotics Navigation

### PR DESCRIPTION
**Pull request by neworderofjamie**
Rather than using its own classes to abstract away the differences between perfect memory and the mushroom body, this refactor wraps the mushroom body model in a class derives from ``BoBRobotics::Navigation::VisualNavigationBase``. Additionally this PR:
* Tidies up the logic using ``BoBRobotics::Common::FSM``
* Tweaked the MB model slightly
* Added functionality to (slowly) render an (ugly) vector field
* Tidied up the bound-calculation in ``BoBRobotics::AntWorld::World`` so it's all done in ``units::length::meter_t``

### CODE REVIEW

File: libantworld/route_ardin.cc [hunk 1]
```diff
@@ -191,6 +192,21 @@ bool RouteArdin::load(const std::string &filename, bool realign)
         }
     }
 
+    // Initialise bounds to limits of underlying data types
+    std::fill_n(&m_MinBound[0], 2, meter_t(std::numeric_limits<meter_t::underlying_type>::max()));
+    std::fill_n(&m_MaxBound[0], 2, meter_t(std::numeric_limits<meter_t::underlying_type>::min()));
```
Comment (alexdewar) [inline libantworld/route_ardin.cc]: I think you might be able to just write:
```
std::fill_n(&m_MinBound[0], 2, std::numeric_limits<meter_t>::max());
```
etc. The units lib has overloads for some numeric_limits functions

File: libantworld/world.cc [hunk 2]
```diff
@@ -124,33 +127,28 @@ bool World::load(const std::string &filename, const GLfloat (&worldColour)[3],
...
 
-        // Convert min/max bounds to metre types (this is admittedly bit gross...)
-        m_MinBoundM[0] = makeM(m_MinBound[0]);
-        m_MinBoundM[1] = makeM(m_MinBound[1]);
-        m_MinBoundM[2] = makeM(m_MinBound[2]);
-        m_MaxBoundM[0] = makeM(m_MaxBound[0]);
-        m_MaxBoundM[1] = makeM(m_MaxBound[1]);
-        m_MaxBoundM[2] = makeM(m_MaxBound[2]);
+        std::cout << "Min: (" << m_MinBound[0] << ", " << m_MinBound[1] << ", " << m_MinBound[2] << ")" << std::endl;
+        std::cout << "Max: (" << m_MaxBound[0] << ", " << m_MaxBound[1] << ", " << m_MaxBound[2] << ")" << std::endl;
```
Comment (alexdewar) [inline libantworld/world.cc]: This is all much tidier!

File: projects/ardin_mb/vector_field.cc [hunk 3]
```diff
@@ -0,0 +1,121 @@
+#include "vector_field.h"
```
Comment (alexdewar) [inline projects/ardin_mb/vector_field.cc]: Another (possibly easier but less flexible) option to doing this in OpenGL would be to use matplotlib-cpp's ``quiver()`` function

**Comment by alexdewar** [comment]
This is all looking good and much tidier :-)

File: projects/ardin_mb/vector_field.cc (hunk 3 above)
Comment (neworderofjamie) [reply projects/ardin_mb/vector_field.cc]: Yeah - the problem is that I want to overlay the result on the OpenGL map view.

I guess theoretically we could draw that whole view using matplotlib but I can see that turning into a shit show.

Comment (alexdewar) [reply projects/ardin_mb/vector_field.cc]: That's true. Plus I suspect you'd have to add your own wrapper function for rendering the triangles anyway.

**Pull request merged**